framework/sessions/__init__.py

```diff
@@ -153,47 +157,55 @@ def before_request():
         return cas.make_response_from_ticket(ticket=ticket, service_url=service_url.url)
 
     # Request Type 2: Basic Auth with username and password in Authorization headers
+    # TODO: Does basic auth need session?
+    #       Does Basic Auth still makes sense for V1/Flask requests?
+    #       Should Basic Auth error on cookie existence?
```

Comment:
As discussed, let's fix the behavior of basic auth by removing the possibility when a cookie is also present.

As for flask/V1 request, many views rely on `auth` that comes from `session()` to identify the user, thus we still need to keep the session creation/usage here.

Reply:
Fixed in https://github.com/CenterForOpenScience/osf.io/pull/10350/commits/f18104cb522191d7a76fd3ad95474ba1c2cd7cda